provider/dnsimple/dnsimple.go

```diff
@@ -118,11 +124,20 @@ func NewDnsimpleProvider(domainFilter endpoint.DomainFilter, zoneIDFilter provid
 		dryRun:       dryRun,
 	}
 
-	whoamiResponse, err := provider.identity.Whoami(context.Background())
-	if err != nil {
-		return nil, err
+	whoamiResponse := &dnsimple.WhoamiResponse{}
+	if !skipWhoami {
+		var err error
+		whoamiResponse, err = provider.identity.Whoami(context.Background())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if dnsimpleAccountId == "" {
+		provider.accountID = int64ToString(whoamiResponse.Data.Account.ID)
+	} else {
+		provider.accountID = dnsimpleAccountId
```

Comment:
```suggestion
	provider.accountID = dnsimpleAccountId
	if provider.accountID == "" {
		whoamiResponse, err := provider.identity.Whoami(context.Background())
		if err != nil {
			return nil, err
		}
		provider.accountID = int64ToString(whoamiResponse.Data.Account.ID)
```

I suggest to remove `skipWhoami bool` and also to avoid `if ... else`. 
If you want to test also this part, you'll need to use mock.

Reply:
Good suggestion - I have implemented this in https://github.com/kubernetes-sigs/external-dns/pull/4274/commits/9c24ecc1084c4eff695da14e1e6b1ef1b9202b77 and https://github.com/kubernetes-sigs/external-dns/pull/4274/commits/660b20b4425da537b324f65c627b0483ea3319d0